src/hotspot/share/services/virtualMemoryTracker.cpp

```diff
@@ -290,7 +290,7 @@ size_t ReservedMemoryRegion::committed_size() const {
 
 void ReservedMemoryRegion::set_flag(MEMFLAGS f) {
   assert((flag() == mtNone || flag() == f),
-         "Overwrite memory type for region [" PTR_FORMAT "-" PTR_FORMAT "), %u->%u.",
+         "Overwrite memory type for region [" INTPTR_FORMAT "-" INTPTR_FORMAT "), %u->%u.",
```

Comment:
Why the explicit change to INTPTR_FORMAT? (real question, I actually do not know the difference to PTR_FORMAT, since their definition seems to be identical)

Reply:
They are defined as same, and p2i:
inline intptr_t p2i(const void * p) {
  return (intptr_t) p;
}
This way I think better use INTPTR_FORMAT, it is consistent with the type here.

Reply:
I agree - semantically we use INTPTR_FORMAT when passing a p2i(x). Though it does beg the question as to why we are applying p2i in this case?